presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -851,6 +852,20 @@ public FeaturesConfig setHistoryCanonicalPlanNodeLimit(int historyCanonicalPlanN
         return this;
     }
 
+    @Min(0)
+    public int getHistoryBasedOptimizerTimeoutInMilliSeconds()
+    {
+        return historyBasedOptimizerTimeoutInMilliSeconds;
+    }
+
+    @Config("optimizer.history-based-optimizer-timeout-in-milli-seconds")
```

Comment:
lets call it "optimizer.history-based-optimizer-timeout". And use Duration instead of int. Config can then read values like "1s" or "100ms"